packages/beacon-node/src/api/impl/lightclient/index.ts

```diff
@@ -16,6 +18,23 @@ export function getLightclientApi(
   // We want some some sort of resistance against this DoS vector.
   const maxGindicesInProof = opts.maxGindicesInProof ?? 512;
 
+  const serializeSSZArray = (chunks: LightClientUpdate[]): Uint8Array => {
```

Comment:
Is there a more ssz way to perform these serialization?

Reply:
using `ssz.altair.LightClientUpdate.serialize()` for each LightClientUpdate is ok
however the way we serialize there may not be correct, it should have some length prefixed data + fork_digest in order for the client to parse:
```
(
                response_chunk_len: Little-endian Uint64 byte length of `response_chunk`
                response_chunk: (
                  context: 4 byte `ForkDigest`
                  payload: SSZ serialized payload bytes
                )
```

we need to have some encode/decode separate functions and make sure it works in unit tests first

Reply:
Hey @tuyennhv, I took some time to dig into `ssz` and found `ListCompositeType` which I was able to use to decode/encode the response. I added test to confirm this. Seems this should be ok? Let me know if you have any other observation/comments.

Reply:
@dadepo  no, that is not correct as ListCompositeType is not mentioned in the spec https://github.com/ethereum/beacon-APIs/blob/master/apis/beacon/light_client/updates.yaml

as noted above and in the spec, it's something like:

```typescript
const result: Buffer[] = [];
for (const update of lightClientUpdates) {
  const payload = ssz.LightClientUpdate.serialize(update);
  const slot = update.attestedHeader.slot;
  const forkName = config.getForkName(slot);
  const context = config.forkName2ForkDigest(forkName);
  const responseChunk = Buffer.concat([context, payload]);
  result.push(Buffer.concat[Buffer.from(varint.encode(responseChunk.length)), responseChunk]);
}

return Buffer.concat(result);
```

Reply:
@tuyennhv Thanks for the clarification. I misinterpreted how the serialisation/deserialisation was supposed to work. I have updated the implementation.

Reply:
oh I misread it, it said `Little-endian Uint64 byte length of response_chunk` so maybe no varint here, just encode/decode with little-endian uint64, in lodestar we have `UintNum64` ssz type so just use it

cc @wemeetagain if you also read the spec the same way